code/game/objects/items/robot/robot_parts.dm

```diff
@@ -114,64 +65,82 @@
 			else
 				user << "<span class='warning'>You need one sheet of metal to start building ED-209!</span>"
 				return
-	else if(istype(W, /obj/item/robot_parts/l_leg))
+	else if(istype(W, /obj/item/bodypart/l_leg/robot))
 		if(src.l_leg)
 			return
 		if(!user.unEquip(W))
 			return
-		W.loc = src
+		W.forceMove(src)
+		W.icon_state = initial(W.icon_state)
+		W.cut_overlays()

```

Comment:
if the bodypart has been used by a mob and then amputated then we need to make sure no previous overlay or icon_state remain.